fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneIndexMaintainer.java

```diff
@@ -236,12 +239,16 @@ private boolean insertField(LuceneDocumentFromRecord.DocumentField field, final
 
     private void writeDocument(@Nonnull List<LuceneDocumentFromRecord.DocumentField> fields, Tuple groupingKey,
                                byte[] primaryKey) throws IOException {
-        final IndexWriter newWriter = directoryManager.getIndexWriter(groupingKey, indexAnalyzer);
+        final List<String> texts = fields.stream()
+                .filter(f -> f.getType().equals(LuceneIndexExpressions.DocumentFieldType.TEXT))
+                .map(f -> (String) f.getValue()).collect(Collectors.toList());
+        final IndexWriter newWriter = directoryManager.getIndexWriter(groupingKey,
+                indexAnalyzerChooser.chooseAnalyzer(texts));
```

Comment:
I think the `FDBDirectoryWrapper` caches the `IndexWriter` for a given `FDBDirecrtory` (affectively assuming that it will always be opened with the same analyzer each time), so I think this will always return an `IndexWriter` with the first `Analyzer` selected on a given transaction.

The manager might need to do something like close the original writer if the analyzer doesn't match (if `Analyzer`s have an appropriate `.equals` implementation) to avoid having multiple writers open at the same time on the same directory

Reply:
Fixed it by having a ```LuceneAnalyzerWrapper``` with a unique identifier for each ```analyzer```. Not the old writer is closed when a new one with a different identifier is needed.